feat: Add frontend documentation to components (#348)
Substra/substra-documentation: docs/source/documentation/frontend/index.rst

********
Frontend
********

The frontend allows you to monitor your assets (compute plans, tasks, datasets, functions) easily through a user interface.

.. _frontend_kubernetes_pods:

Kubernetes pods
===============

frontend
    A single pod managing the frontend. 

.. _frontend_communication:

Communication
=============

> The Backend should be able to reach its Backend through the REST API.

Reviewer: ```suggestion
The frontend should be able to reach its backend through the REST API.
```

> The access to the API is secured through the use of Json Web Tokens (JWT), which are stored through cookies. Each backend pods has its own tokens, so when working with different backends or restarting pods, it might be necessary to delete related cookies (namely signature, refresh and header.payload) so a new JWT can be created. Otherwise this could block you from logging into the Frontend.   

Reviewer: ```suggestion
The access to the API is secured through the use of Json Web Tokens (JWT), which are stored through cookies. Each backend pods has its own tokens, so when working with different backends or restarting pods, it might be necessary to delete related cookies (namely signature, refresh and header.payload) so a new JWT can be created. Otherwise this could block you from logging into the frontend.   
```

********
Frontend
********

> The frontend (also named web application for the end-users) allows you to monitor your assets (compute plans, tasks, datasets, functions) easily through a user interface. It is mainly a read-only interface:  you will need to use the python library to register data or to launch computation. However there are a few actions that are doable with the frontend, for instance: cancelling compute plan, and managing users as well as API tokens.

Reviewer: ```suggestion
The frontend (also named web application for the end-users) allows you to monitor your assets (compute plans, tasks, datasets, functions) easily through a user interface. It is mainly a read-only interface:  you will need to use the Python library to register data or to launch computation. However there are a few actions that are doable with the frontend, for instance: cancelling compute plans, managing users and creating API tokens.
```

.. _frontend_kubernetes_pods:

Kubernetes pods
===============

frontend
    A single pod managing the frontend. 

.. _frontend_communication:

Communication
=============

The frontend should be able to reach its backend through the REST API.

> The access to the API is secured through the use of Json Web Tokens (JWT), which are stored through cookies. Each backend pods has its own tokens, so when working with different backends or restarting pods, it might be necessary to delete related cookies (namely signature, refresh and header.payload) so a new JWT can be created. Otherwise this could block you from logging into the frontend.   

Reviewer: ```suggestion
The access to the API is secured through the use of Json Web Tokens (JWT), which are stored through cookies. Each backend pod has its own tokens, so when working with different backends or restarting pods, it might be necessary to delete related cookies (namely signature, refresh and header.payload) so a new JWT can be created. Otherwise this could block you from logging into the frontend.   
```

Reviewer: I am not sure though.

Author: I will go with `server pod` to be clearer as it is the one exposing the REST API.
